poh/benches/transaction_recorder.rs

```diff
@@ -0,0 +1,118 @@
+use {
+    criterion::{criterion_group, criterion_main, Criterion},
+    solana_clock::UPDATED_HASHES_PER_TICK6,
+    solana_hash::Hash,
+    solana_keypair::Keypair,
+    solana_ledger::{
+        blockstore::Blockstore, genesis_utils::create_genesis_config,
+        get_tmp_ledger_path_auto_delete, leader_schedule_cache::LeaderScheduleCache,
+    },
+    solana_poh::{
+        poh_recorder::PohRecorder,
+        poh_service::{PohService, DEFAULT_HASHES_PER_BATCH, DEFAULT_PINNED_CPU_CORE},
+        transaction_recorder::TransactionRecorder,
+    },
+    solana_poh_config::PohConfig,
+    solana_pubkey::Pubkey,
+    solana_runtime::{bank::Bank, installed_scheduler_pool::BankWithScheduler},
+    solana_transaction::versioned::VersionedTransaction,
+    std::{
+        sync::{atomic::AtomicBool, Arc, RwLock},
+        time::{Duration, Instant},
+    },
+};
+
+fn bench_record_transactions(c: &mut Criterion) {
+    const NUM_TRANSACTIONS: u64 = 16;
+    const NUM_BATCHES: u64 = 32;
+
+    // Setup the PohService.
+    let mut genesis_config_info = create_genesis_config(2);
+    genesis_config_info.genesis_config.ticks_per_slot = 64;
+    genesis_config_info.genesis_config.poh_config = PohConfig {
+        target_tick_duration: Duration::from_millis(400),
+        target_tick_count: None,
+        hashes_per_tick: Some(UPDATED_HASHES_PER_TICK6),
+    };
+    let exit = Arc::new(AtomicBool::new(false));
+    let bank = Arc::new(Bank::new_for_tests(&genesis_config_info.genesis_config));
+    let ledger_path = get_tmp_ledger_path_auto_delete!();
+    let blockstore = Arc::new(
+        Blockstore::open(ledger_path.path()).expect("Expected to be able to open database ledger"),
+    );
+    let (mut poh_recorder, _entry_receiver) = PohRecorder::new(
+        bank.tick_height(),
+        bank.last_blockhash(),
+        bank.clone(),
+        Some((0, 0)),
```

Comment:
I'm assuming we don't expect slot transitions to happen during this bench? So presumably leader slot range doesn't matter